tests/providers/amazon/aws/hooks/test_glacier.py

```diff
@@ -0,0 +1,173 @@
+            self.hook = GlacierHook(aws_conn_id="aws_default")
+
+    @mock.patch("airflow.providers.amazon.aws.hooks.glacier.GlacierHook.get_conn")
+    def test_retrieve_inventory_should_be_called_once_with(self, mock_conn):
+        # when
+        self.hook.retrieve_inventory(VAULT_NAME)
+        # then
+        mock_conn.assert_called_once_with()
+
+    @mock.patch("airflow.providers.amazon.aws.hooks.glacier.GlacierHook.get_conn")
+    def test_retrieve_inventory_should_return_job_id(self, mock_conn):
+        # Given
+        job_id = {"jobId": "1234abcd"}
+        # when
+        mock_conn.return_value.initiate_job.return_value = job_id
+        result = self.hook.retrieve_inventory(VAULT_NAME)
+        # then
+        self.assertEqual(job_id, result)
+
+    @mock.patch(
+        "airflow.providers.amazon.aws.hooks.glacier.GlacierHook.get_conn", side_effect=[ClientError({}, {})]
+    )
+    def test_retrieve_inventory_should_raise_client_error_when_error_occurred(self, mock_conn):
+        with pytest.raises(ClientError):
+            self.hook.retrieve_inventory(VAULT_NAME)
+
+    @mock.patch("airflow.providers.amazon.aws.hooks.glacier.GlacierHook.get_conn")
+    def test_retrieve_inventory_should_log_mgs(self, mock_conn):
+        # given
+        job_id = {"jobId": "1234abcd"}
+
+        # when
+        with LogCapture() as log:
+            mock_conn.return_value.initiate_job.return_value = job_id
+            self.hook.retrieve_inventory(VAULT_NAME)
+
+            # then
+            log.check(
+                (
+                    'airflow.providers.amazon.aws.hooks.glacier.GlacierHook',
+                    'INFO',
+                    f"Retrieving inventory for vault: {VAULT_NAME}",
+                ),
+                (
+                    'airflow.providers.amazon.aws.hooks.glacier.GlacierHook',
+                    'INFO',
+                    f"Initiated inventory-retrieval job for: {VAULT_NAME}",
+                ),
+                (
+                    'airflow.providers.amazon.aws.hooks.glacier.GlacierHook',
+                    'INFO',
+                    f"Retrieval Job ID: {job_id.get('jobId')}",
+                ),
+            )
+
+    @mock.patch("airflow.providers.amazon.aws.hooks.glacier.GlacierHook.get_conn")
+    def test_retrieve_inventory_results_should_be_called_once_with(self, mock_conn):
+        # when
+        self.hook.retrieve_inventory_results(vault_name=VAULT_NAME, job_id=JOB_ID)
+        # then
+        mock_conn.assert_called_once_with()
+
+    @mock.patch("airflow.providers.amazon.aws.hooks.glacier.GlacierHook.get_conn")
+    def test_retrieve_inventory_results_should_return_(self, mock_conn):
+        # when
+        mock_conn.return_value.get_job_output.return_value = RESPONSE_BODY
+        response = self.hook.retrieve_inventory_results(VAULT_NAME, JOB_ID)
+        # then
+        self.assertEqual(response, RESPONSE_BODY)
```

Comment:
Personally I would combine those two test into one

Reply:
Frankly speaking, I am not a big fan of tests with multiple assertion within one test method.

Reply:
I would say there no need for two assertions. From logical point of view if `test_retrieve_inventory_results_should_return_` works as expected it also means that the method was called

Plus why the test name ends with `_`? 😉

Reply:
The `_` is my mistake. ;)

Reply:
Yes, I agree but we do not know how many times method was called. :)
I will move assertion to the second method.